extension/cjk.go

```diff
@@ -9,11 +9,30 @@ import (
 // A CJKOption sets options for CJK support mostly for HTML based renderers.
 type CJKOption func(*cjk)
 
+// A EastAsianLineBreaksStyle is a style of east asian line breaks.
+type EastAsianLineBreaksStyle int
+
+const (
+	// EastAsianLineBreaksStyleSimple is a style where soft line breaks are ignored
+	// if both sides of the break are east asian wide characters.
+	EastAsianLineBreaksStyleSimple EastAsianLineBreaksStyle = iota
+	// EastAsianLineBreaksCSS3Draft is a style where soft line breaks are ignored
+	// even if only one side of the break is an east asian wide character.
+	EastAsianLineBreaksCSS3Draft
```

Comment:
Do you have a better name than `EastAsianLineBreaksCSS3Draft`?

Reply:
It seems your implementation does not satisfy CSS3 draft rules. We may have choices...

1. Implement CSS3 Draft rules and leave this name as it is
2. Change this name to represent your implementation

I prefer 1. 

- Implement CSS text level 3 draft rules.
- Implement additional enhancement(i.e. resolves https://github.com/w3c/csswg-drafts/issues/5086)
- Write README like 'This option implements CSS text level3 [Segment Break Transformation Rules](https://drafts.csswg.org/css-text-3/#line-break-transform) with some enhancements'

Reply:
Thank you for your thorough feedback.

I’m on board with option 1 and will give the CSS text level 3 rules and additional enhancements a shot. Admittedly, I'm not a pro with this CSS issue, so while I'll do my best, I might miss some nuances. Any extra guidance or pointers while I work through this would be awesome!

Will keep you posted on the progress. Talk to you soon!